runtime/src/bank/fee_distribution.rs

```diff
@@ -161,53 +212,40 @@ impl Bank {
-                        || (self.prevent_rent_paying_rent_recipients()
-                            && !rent_state_transition_allowed)
-                    {
-                        // overflow adding lamports or resulting account is not rent-exempt
-                        self.capitalization.fetch_sub(rent_to_be_paid, Relaxed);
-                        error!(
-                            "Burned {} rent lamports instead of sending to {}",
-                            rent_to_be_paid, pubkey
-                        );
-                        datapoint_error!(
-                            "bank-burned_rent",
-                            ("slot", self.slot(), i64),
-                            ("num_lamports", rent_to_be_paid, i64)
-                        );
-                    } else {
-                        self.store_account(&pubkey, &account);
-                        rewards.push((
-                            pubkey,
-                            RewardInfo {
-                                reward_type: RewardType::Rent,
-                                lamports: rent_to_be_paid as i64,
-                                post_balance: account.lamports(),
-                                commission: None,
-                            },
-                        ));
+                    let check_account_owner = self.validate_fee_collector_account();
+                    let check_rent_paying = self.prevent_rent_paying_rent_recipients();
+                    match self.deposit_fees(
+                        &pubkey,
+                        rent_to_be_paid,
+                        DepositFeeOptions {
+                            check_account_owner,
+                            check_rent_paying,
+                        },
+                    ) {
+                        Ok(post_balance) => {
+                            rewards.push((
+                                pubkey,
+                                RewardInfo {
+                                    reward_type: RewardType::Rent,
+                                    lamports: rent_to_be_paid as i64,
+                                    post_balance,
+                                    commission: None,
+                                },
+                            ));
+                        }
+                        Err(_) => {
+                            // overflow adding lamports or resulting account is invalid
+                            self.capitalization.fetch_sub(rent_to_be_paid, Relaxed);
+                            error!(
+                                "Burned {} rent lamports instead of sending to {}",
+                                rent_to_be_paid, pubkey
+                            );
+                            datapoint_error!(
+                                "bank-burned_rent",
+                                ("slot", self.slot(), i64),
+                                ("num_lamports", rent_to_be_paid, i64)
+                            );
```

Comment:
Do we maybe want to back down these, or at least the log? It's not really the kind of thing a node operator would interpret as an error.

Reply:
Yeah good idea. I think that if a validator is burning its own fees it should see an error but it won't care about other validators. Do you care if I remove the datapoint?

Reply:
I don't know of a need for the datapoint and am personally fine with its removal, but that might be a good question for @jeffwashington or @HaoranYi

Reply:
@HaoranYi is there a specific reason why you would like to track this datapoint? I can't think of any reason personally.

Reply:
It gives us the visibility of how much fees are burned in the network. The amount of the fees burned should be very small.  To me, being able to run such a query and verify the amount of fees burned, could be a useful sanity check.

Reply:
Good point, I'll add it back in, thanks

Reply:
I added the datapoint back in but downgraded it to "warn"